server/remote_cache/cachetools/cachetools.go

```diff
@@ -462,6 +462,17 @@ func ReadProtoFromCache(ctx context.Context, cache interfaces.Cache, r *digest.C
 	return proto.Unmarshal(data, out)
 }
 
+func ReadProtoFromActionCache(ctx context.Context, cache interfaces.Cache, r *digest.ACResourceName, out proto.Message) error {
```

Comment:
This looks basically the same as ReadProtoFromCache, except the error status. Is it possible to make them both call a shared, private function?

Reply:
MissingDigestError wraps the `NotFound` into a `FailedPrecondition` error, and I assume that's because of some Bazel trivia.  But it looks like the only callers of `ReadProtoFromCache` right now don't actually do any checking at all on the error type, so I suppose I can just change both to always return an unwrapped error.  @tylerwilliams wdyt?

Reply:
yeah that's right (spec is [here](https://github.com/bazelbuild/remote-apis/blob/2721568dea746d81a98a116ee222c8da50bbcf5c/build/bazel/remote/execution/v2/remote_execution.proto#L103-L112) if you're curious)

It looks like the returned error is bubbled up through ```func (s *ExecutionServer) Execute(req *repb.ExecuteRequest, stream repb.Execution_ExecuteServer) error {``` so I assume it does make it back to bazel. You could remove the MissingDigest wrapping from this function in cachetools and add it in `fetchActionAndCommand` if you wanted to reuse this?

Reply:
that probably is more correct -- since only Execute cares about adding that MissingDigest precondition